helpers/deployContracts.ts

```diff
@@ -1,31 +1,277 @@
+      pluginRepoFactory,
+      allowListPluginSetup.address,
+      "AllowlistVoting",
+      [1, 0, 0],
+      owner
+    );
+    const allowListRepo = pluginRepo_Factory
+      .connect(owner)
+      .attach(allowListRepoAddr);
+
+    // send ETH to hardcoded wallet in tests
+    await owner.sendTransaction({
+      to: WALLET_ADDRESS,
+      value: ethers.utils.parseEther("50.0"),
+    });
+
+    return {
+      daoFactory,
+      erc20Repo,
+      erc20PluginSetup,
+      allowListRepo,
+      allowListPluginSetup,
+    };
+  } catch (e) {
+    throw e;
+  }
+}
+
+async function deployPlugin(
+  pluginRepoFactory: aragonContracts.PluginRepoFactory,
+  setupAddr: string,
+  name: string,
+  version: [BigNumberish, BigNumberish, BigNumberish],
+  owner: Signer
+) {
+  const repoaddr = await pluginRepoFactory.callStatic.createPluginRepoWithVersion(
+    name,
+    version,
+    setupAddr,
+    "0x",
+    await owner.getAddress()
+  );
+  const tx = await pluginRepoFactory.createPluginRepoWithVersion(
+    name,
+    version,
+    setupAddr,
+    "0x",
+    await owner.getAddress()
+  );
+  await tx.wait();
+  return repoaddr;
+}
+
+async function deployEnsContracts(owner: Signer) {
+  try {
+    const registryFactory = new ethers.ContractFactory(
+      ENSRegistry.abi,
+      ENSRegistry.bytecode
+    );
+    const publicResolverFactory = new ethers.ContractFactory(
+      PublicResolver.abi,
+      PublicResolver.bytecode
+    );
+
+    const registry = await registryFactory.connect(owner).deploy();
+    await registry.deployed();
+
+    const publicResolver = await publicResolverFactory
+      .connect(owner)
+      .deploy(
+        registry.address,
+        ethers.constants.AddressZero,
+        ethers.constants.AddressZero,
+        ethers.constants.AddressZero
+      );
+    await publicResolver.deployed();
+
+    await registerEnsName(
+      "",
+      "eth",
+      registry,
+      await owner.getAddress(),
+      publicResolver.address
+    );
+    return {ensRegistry: registry, ensResolver: publicResolver};
+  } catch (e) {
+    throw e;
+  }
+}
+
+export async function registerEnsName(
+  tld: string,
+  name: string,
+  registry: Contract,
+  owner: string,
+  resolver: string
+) {
+  try {
+    await registry.setSubnodeRecord(
+      tld !== "" ? ethers.utils.namehash(tld) : ethers.constants.HashZero,
+      ethers.utils.id(name),
+      owner,
+      resolver,
+      0
+    );
+  } catch (e) {
+    throw e;
+  }
+}
+
+export async function createDAO(
+  daoFactory: aragonContracts.DAOFactory,
+  daoSettings: aragonContracts.DAOFactory.DAOSettingsStruct,
+  pluginSettings: aragonContracts.DAOFactory.PluginSettingsStruct[]
+) {
+  let daoAddress = await daoFactory.callStatic.createDao(
+    daoSettings,
+    pluginSettings
+  );
+  await daoFactory.createDao(daoSettings, pluginSettings);
+  return daoAddress;
+}
+
+export async function createAllowlistDAO(deployment: Deployment, name: string) {
```

Comment:
Can we have the ERC20 counterpart? Otherwise, let's open a follow-up task

Reply:
we can but it wasn't needed in the current tests so there is none in the helper functions.
I'll create the function anyways to test the ERC20Voting createProposal stuff.